api/src/main/java/org/opennms/newts/api/query/StandardAggregationFunctions.java

```diff
@@ -69,6 +71,37 @@ public Double apply(Collection<Double> input) {
             return min;
         }
 
+    },
+    P95 {
+
+        @Override
+        public Double apply(Collection<Double> input) {
+            return ntile(0.95, input);
+        }
+
+    },
+    P99 {
+
+        @Override
+        public Double apply(Collection<Double> input) {
+            return ntile(0.99, input);
+        }
+
+    };
+
+    private static Double ntile(double n, Collection<Double> input) {
+        List<Double> sorted = new ArrayList<Double>();
+        for (Double item : input) {
+            if (!Double.isNaN(item)) {
+                sorted.add(item);
+            }
+        }
+        if (sorted.isEmpty()) {
+            return Double.MAX_VALUE;
```

Comment:
I'm wondering if MAX_VALUE is the right thing here? Other aggregations use NaN in such a case.